R/serializer.R

```diff
@@ -652,8 +652,8 @@ createGraphicsDevicePromiseDomain <- function(which = dev.cur()) {
       force(onFulfilled)
       function(...) {
         old <- dev.cur()
-        dev.set(which)
-        on.exit(dev.set(old))
+        dev_set(which)
+        on.exit(dev_set(old))
```

Comment:
I *think* what you want is closer to:

```r
old <- dev.cur()
on.exit({
  if (old > 1) dev.set(old) else dev.off()
})
dev.set(which)
```

Reply:
Dev off is handled elsewhere. This is for handling the graphics device when changing promise domains

Reply:
When I do call dev off, I am telling which device to turn off

Reply:
Ok, maybe you could just remove the `on.exit()` then? Let's talk through it later this week